docs/source/tutorials/transfer_learning/transfer_learning_example.py

```diff
@@ -0,0 +1,247 @@
+
+
+def extract_transferable_evaluations(df, metric, config_space):
+    """
+    Take a dataframe from a tuner run, filter it and generate
+    TransferLearningTaskEvaluations from it
+    """
+
+    # Filter out nan values, i.e. make sure we only use completed runs
+    filter_df = df[~pd.isna(df[metric])].reset_index()
+
+    return TransferLearningTaskEvaluations(
+        configuration_space=config_space,
+        hyperparameters=filter_df[config_space.keys()],
+        objectives_names=[metric],
+        # objectives_evaluations need to be of shape (num_evals, num_seeds, num_fidelities, num_objectives)
+        # We only have one seed, fidelity and objective
+        objectives_evaluations=np.array(filter_df[metric], ndmin=4).T,
+    )
+
+
+def run_scheduler_on_task(entry_point, scheduler, max_trials):
+    """
+    Take a scheduler and run it for max_trials on the backend specified by entry_point
+    Return a dataframe of the optimisation results
+    """
+    tuner = Tuner(
+        trial_backend=LocalBackend(entry_point=str(entry_point)),
+        scheduler=scheduler,
+        stop_criterion=StoppingCriterion(max_num_trials_finished=max_trials),
+        n_workers=4,
+        sleep_time=0.001,
+    )
+    tuner.run()
+
+    return tuner.tuning_status.get_dataframe()
+
+
+def init_scheduler(
+    scheduler_str, max_steps, seed, mode, metric, transfer_learning_evaluations
+):
+    """
+    Initialise the scheduler
+    """
+    kwargs = {
+        "metric": metric,
+        "config_space": height_config_space(max_steps=max_steps),
+        "mode": mode,
+        "random_seed": seed,
+    }
+    kwargs_w_trans = copy.deepcopy(kwargs)
+    kwargs_w_trans["transfer_learning_evaluations"] = transfer_learning_evaluations
+
+    if scheduler_str == "BayesianOptimization":
+        return BayesianOptimization(**kwargs)
+
+    if scheduler_str == "ZeroShotTransfer":
+        return ZeroShotTransfer(use_surrogates=True, **kwargs_w_trans)
+
+    if scheduler_str == "Quantiles":
+        return FIFOScheduler(
+            points_to_evaluate=[],
+            searcher=QuantileBasedSurrogateSearcher(**kwargs_w_trans),
+            **kwargs,
+        )
+
+    if scheduler_str == "BoundingBox":
+        kwargs_sched_fun = {key: kwargs[key] for key in kwargs if key != "config_space"}
+        kwargs_w_trans[
+            "scheduler_fun"
+        ] = lambda new_config_space, mode, metric: FIFOScheduler(
+            new_config_space,
+            points_to_evaluate=[],
+            searcher="bayesopt",
+            **kwargs_sched_fun,
+        )
+        del kwargs_w_trans["random_seed"]
+        return BoundingBox(**kwargs_w_trans)
+    raise ValueError("scheduler_str not recognised")
+
+
+if __name__ == "__main__":
+
+    mode = "min"
+    max_trials = 10
+    np.random.seed(1)
+    # Use train_height backend for our tests
+    entry_point = "../../../../examples/training_scripts/height_example/train_height.py"
```

Comment:
Best to make this relative to this file here, using `Path(__file__).parent`, as in the other examples.

Reply:
Done.